### PR TITLE
fix(web): general app layout updates for small and bigger screens

### PR DESCRIPTION
**Pull request by LetItRock**
### What change does this PR introduce?

**Sorry about the big PR, but I've noticed so many small issues...**

**Also want to ask you to do the local dev testing for this PR, because it might be hard to understand from the code why I've changed something.**

In this PR I've fixed:
- the header was not sticky at the top
- the double scrollbar was shown on smaller screens
- the side nav was too big on smaller screens, which created layout problems
- the get started footer was not sticky to the bottom of the page
- the digest tour component was not visible on smaller screens
- many more

### Other information (Screenshots)

I've touched too many places in the app so it's hard to put all the screenshots.

### CODE REVIEW

File: apps/web/src/components/layout/components/SideNav.tsx [hunk 1]
```diff
@@ -193,45 +193,39 @@ export function SideNav({}: Props) {
           </Popover.Dropdown>
         </Popover>
         <NavMenu menuItems={menuItems} />
-      </Navbar.Section>
-      <Navbar.Section mt={15}>
-        <Navbar.Section>
```
Comment (LetItRock) [inline apps/web/src/components/layout/components/SideNav.tsx:198]: this nesting was not needed

File: apps/web/src/components/layout/components/SideNav.tsx [hunk 2]
```diff
@@ -137,17 +137,16 @@ export function SideNav({}: Props) {
...
       sx={{
         position: 'sticky',
         top: 0,
         zIndex: 'auto',
         backgroundColor: 'transparent',
         borderRight: 'none',
-        paddingRight: 0,
         width: '300px',
-        height: 'max-content',
+        minHeight: '100vh',
```
Comment (LetItRock) [inline apps/web/src/components/layout/components/SideNav.tsx:147]: the navigation sidebar was causing layout problems on smaller screens like
Screenshot 2023-08-11 at 01 02 39

File: apps/web/src/design-system/navigation/NavMenu.styles.ts [hunk 3]
```diff
@@ -39,14 +39,23 @@ export default createStyles((theme: MantineTheme) => {
...
       },
+
+      '&:last-of-type': {
+        marginBottom: '0px',
+      },
+
+      '@media (min-width: 1440px)': {
+        margin: '15px 0px',
+        height: '50px',
+      },
```
Comment (LetItRock) [inline apps/web/src/design-system/navigation/NavMenu.styles.ts:50]: basically here adjusted the styles for the navigation on smaller screens

File: apps/web/src/design-system/sidebar/Sidebar.tsx [hunk 4]
```diff
@@ -108,7 +108,7 @@ export const Sidebar = ({
       position="right"
       styles={{
         drawer: {
-          width: isExpanded ? `calc(100% - ${NAVIGATION_WIDTH + PAGE_MARGIN}px)` : COLLAPSED_WIDTH,
+          width: isExpanded ? `calc(100% - ${NAVIGATION_WIDTH}px)` : COLLAPSED_WIDTH,
```
Comment (LetItRock) [inline apps/web/src/design-system/sidebar/Sidebar.tsx:111]: updated the `Sidebar` width calculations for the expended state

File: apps/web/src/pages/activities/services/chart-bar/option.service.ts [hunk 5]
```diff
@@ -83,6 +83,7 @@ function getTooltipConfiguration() {
         tooltipEl.id = 'chartjs-tooltip';
         tooltipEl.innerHTML = '<table></table>';
         document.body.appendChild(tooltipEl);
+        tooltipEl.style.position = 'absolute';
```
Comment (LetItRock) [inline apps/web/src/pages/activities/services/chart-bar/option.service.ts:86]: this was a super weird issue when you navigate to the activity feed, we do show there a tooltip on the bar chart and it's created and inserted into the dom, and if the user navigates to the workflow page and the tour hints were shown it caused the layout changes... in the screenshot you can see that the hint caused layout shift but also when I fixed that the tooltip also cause that shift as it takes the space

Screenshot 2023-08-11 at 01 07 43

File: apps/web/src/pages/quick-start/components/layout/BodyLayout.tsx [hunk 6]
```diff
@@ -25,11 +24,10 @@ export function BodyLayout({ children }: { children: React.ReactNode }) {
 }
 
 const StyledBody = styled.div`
-  display: block;
-  flex: auto;
-  margin-bottom: ${FOOTER_HEIGHT}px;
+  height: calc(100vh - 180px);
   padding-top: 25px;
   padding-bottom: 25px;
+  overflow-y: auto;
```
Comment (LetItRock) [inline apps/web/src/pages/quick-start/components/layout/BodyLayout.tsx:30]: get started page body scrollable on smaller screens and footer sticks to the bottom

File: apps/web/src/pages/quick-start/components/layout/FooterLayout.tsx [hunk 7]
```diff
@@ -63,24 +63,14 @@ const RightCol = styled(Grid.Col)`
 `;
 
 const FooterWrapper = styled.div`
-  position: absolute;
-  bottom: 0;
-  right: 0;
-  left: 0;
```
Comment (LetItRock) [inline apps/web/src/pages/quick-start/components/layout/FooterLayout.tsx:69]: footer doesn't need to be positioned absolute

File: apps/web/src/pages/quick-start/components/layout/GetStartedLayout.tsx [hunk 8]
```diff
@@ -48,7 +48,9 @@ export function GetStartedLayout({ children, footer, header }: IGetStartedLayout
     <>
       <PageContainer style={{ display: 'flex' }}>
         <PageWrapper>
-          <HeaderLayout>{header}</HeaderLayout>
+          <HeaderLayout>
+            <Title>Get started</Title>
+          </HeaderLayout>
```
Comment (LetItRock) [inline apps/web/src/pages/quick-start/components/layout/GetStartedLayout.tsx:53]: all the get started pages should have same title

File: apps/web/src/pages/templates/components/DeleteStepRow.tsx [hunk 9]
```diff
@@ -20,15 +20,7 @@ export const DeleteStepRow = () => {
...
 
   return (
-    <Group
-      position="apart"
-      sx={{
-        position: 'absolute',
-        bottom: 24,
-        left: 24,
-        right: 24,
-      }}
```
Comment (LetItRock) [inline apps/web/src/pages/templates/components/DeleteStepRow.tsx:30]: fixed some weird styles with the delete button in the step editor

File: apps/web/src/pages/templates/components/SubPageWrapper.tsx [hunk 10]
```diff
@@ -23,17 +24,19 @@ export const SubPageWrapper = ({
   return (
     <div
       style={{
+        display: 'flex',
+        flexDirection: 'column',
+        gap: 24,
```
Comment (LetItRock) [inline apps/web/src/pages/templates/components/SubPageWrapper.tsx:29]: this is the container for the workflow step sidebar and email/in-app editors... the items should be spaced equally

File: apps/web/src/pages/templates/components/SubPageWrapper.tsx [hunk 11]
```diff
@@ -23,17 +24,19 @@ export const SubPageWrapper = ({
...
         backgroundColor: colorScheme === 'dark' ? colors.B17 : colors.white,
         borderRadius: '0 7px 7px 0 ',
-        height: '100%',
         width: 460,
         position: 'relative',
         ...style,
       }}
       data-test-id="step-page-wrapper"
     >
-      <div style={{ display: 'flex', marginBottom: 24 }}>
```
Comment (LetItRock) [inline apps/web/src/pages/templates/components/SubPageWrapper.tsx:36]: then removed many margins for items as they were not consistent

File: apps/web/src/pages/templates/editor/DigestWorkflowTourTooltip.tsx [hunk 12]
```diff
@@ -25,11 +25,30 @@ const DESCRIPTION = [
...
+  @media screen and (min-width: 1440px) {
+    padding: 32px;
+
+    [data-tooltip-icon] {
+      width: 120px;
+      height: 120px;
+      margin-top: 40px;
+      margin-bottom: 40px;
+    }
+  }
```
Comment (LetItRock) [inline apps/web/src/pages/templates/editor/DigestWorkflowTourTooltip.tsx:51]: Fixed the styles for the hint component on smaller screens. The problem was that it was causing layout shifts and it wasn't accessible.

File: apps/web/src/pages/templates/workflow/DelayMetadata.tsx [hunk 13]
```diff
@@ -21,48 +21,34 @@ export const DelayMetadata = ({ control, index }) => {
 
   return (
     <>
-      <div
-        style={{
-          marginBottom: '15px',
```
Comment (LetItRock) [inline apps/web/src/pages/templates/workflow/DelayMetadata.tsx:26]: removed this div

File: apps/web/src/pages/templates/workflow/WorkflowEditor.tsx [hunk 14]
```diff
@@ -241,19 +240,12 @@ const WorkflowEditor = () => {
             />
           </div>
 
-          <div
-            style={{
-              width: 'auto',
-              minHeight: '600px',
```
Comment (LetItRock) [inline apps/web/src/pages/templates/workflow/WorkflowEditor.tsx:247]: removed div

File: apps/web/src/pages/tenants/components/list/TenantsList.tsx [hunk 15]
```diff
@@ -30,13 +30,7 @@ export const TenantsList = ({
   }
 
   return (
-    <PageContainer
-      style={{
-        position: 'relative',
-        overflow: 'hidden',
-      }}
-      title="Tenants"
-    >
```
Comment (LetItRock) [inline apps/web/src/pages/tenants/components/list/TenantsList.tsx:39]: this was duplicated on the tenants page

**Review by scopsy — APPROVED** [review]

**Pull request merged**